processor/src/trace.rs

```diff
@@ -1,27 +1,33 @@
 use super::{
-    AuxiliaryTableTrace, Bitwise, Felt, FieldElement, Hasher, Memory, Process, StackTrace,
-    AUXILIARY_TABLE_WIDTH, MIN_STACK_DEPTH,
+    AuxTableTrace, Bitwise, Digest, Felt, FieldElement, Hasher, Memory, Process, StackTopState,
+    StackTrace, SysTrace,
 };
 use core::slice;
+use vm_core::{
+    StarkField, AUX_TRACE_OFFSET, AUX_TRACE_RANGE, AUX_TRACE_WIDTH, STACK_TRACE_OFFSET,
+    STACK_TRACE_RANGE, STACK_TRACE_WIDTH, SYS_TRACE_OFFSET, SYS_TRACE_RANGE, TRACE_WIDTH,
+};
 use winterfell::Trace;
 
 // VM EXECUTION TRACE
 // ================================================================================================
 
-/// TODO: for now this consists only of stack trace and auxiliary traces, but will need to
-/// include decoder trace, etc.
+/// TODO: for now this consists only of system register trace, stack trace, and auxiliary table
+/// trace, but will also need to include decoder trace and range checker trace.
 pub struct ExecutionTrace {
     meta: Vec<u8>,
+    system: SysTrace,
     stack: StackTrace,
-    #[allow(dead_code)]
-    aux_table: AuxiliaryTableTrace,
+    aux_table: AuxTableTrace,
+    // TODO: program hash should be retrieved from decoder trace, but for now we store it explicitly
+    program_hash: Digest,
```

Comment:
Let's come back to this later (I'll make a note in the code). My hesitation is that without "trace" part the word "auxiliary" doesn't stand on its own. This may be fine - but I'd like to take another look at it (outside of this PR).

Reply:
I'm assuming you meant "without "table"", rather than trace, but yes, revisiting this later separately sounds fine - or just leaving it like this. It was just a thought that came up while reading through all of the different related things